### PR TITLE
Additional listener should inherit the configured authentication method

### LINKED ISSUE #7531: [Enhancement]: additional listener should inherit the configured authentication method
### Module

Redpanda

### Proposal

With the current [redpanda.yaml](https://github.com/testcontainers/testcontainers-java/blob/d99b472caab6081ecb9570d340c4213cc51941cd/modules/redpanda/src/main/resources/testcontainers/redpanda.yaml.ftl#L24-L28) template, additional listener are added as follow:

```ftl
<#list kafkaApi.listeners as listener>
    - address: 0.0.0.0
      name: ${listener.address}
      port: ${listener.port}
</#list>
```

However if authentication is turned on, then using such listeners would fail because no authentication is configured. Ideally any new additional listener should inherit the defined `authentication_method` or - but that would be a little bit more convoluted - allow the user to also define the authentication method one adding a listener, i.e.

```java
RedpandaContainer redpanda = new RedpandaContainer("docker.redpanda.com/redpandadata/redpanda:v23.1.7")
    .withListener(() -> "redpanda:19092?auth=sasl")
    .withNetwork(network);
```

I can provide a PR if this enhancement is accepted.

### PR DESCRIPTION
**Pull request by lburgazzoli**
This PR implement the enhancement proposed in #7531 by setting the `authentication_method` for the additional listeners to the authentication method configured when sasl is enabled.

Fixes #7531

### CODE REVIEW

File: modules/redpanda/src/test/java/org/testcontainers/redpanda/RedpandaContainerTest.java [hunk 1]
```diff
@@ -138,6 +139,90 @@ public void testUsageWithListener() throws Exception {
...
 
+    @Test
+    public void testUsageWithListenerAndSasl() throws Exception {
+        final String username = "panda";
+        final String password = "pandapass";
+        final String algorithm = "SCRAM-SHA-256";
+
+        try (
+            Network network = Network.newNetwork();
+            // registerListener {
```
Comment (eddumelendez) [inline modules/redpanda/src/test/java/org/testcontainers/redpanda/RedpandaContainerTest.java:150]:
```suggestion
```

File: modules/redpanda/src/test/java/org/testcontainers/redpanda/RedpandaContainerTest.java [hunk 2]
```diff
@@ -138,6 +139,90 @@ public void testUsageWithListener() throws Exception {
...
+            RedpandaContainer redpanda = new RedpandaContainer("docker.redpanda.com/redpandadata/redpanda:v23.1.7")
+                .enableAuthorization()
+                .enableSasl()
+                .withSuperuser("panda")
+                .withListener(() -> "my-panda:29092")
+                .withNetwork(network);
+            // }
+            // createKCatContainer {
```
Comment (eddumelendez) [inline modules/redpanda/src/test/java/org/testcontainers/redpanda/RedpandaContainerTest.java:158]:
```suggestion
```

File: modules/redpanda/src/test/java/org/testcontainers/redpanda/RedpandaContainerTest.java [hunk 3]
```diff
@@ -138,6 +139,90 @@ public void testUsageWithListener() throws Exception {
...
+            GenericContainer<?> kcat = new GenericContainer<>("confluentinc/cp-kcat:7.4.1")
+                .withCreateContainerCmdModifier(cmd -> {
+                    cmd.withEntrypoint("sh");
+                })
+                .withCopyToContainer(Transferable.of("Message produced by kcat"), "/data/msgs.txt")
+                .withNetwork(network)
+                .withCommand("-c", "tail -f /dev/null")
+            // }
```
Comment (eddumelendez) [inline modules/redpanda/src/test/java/org/testcontainers/redpanda/RedpandaContainerTest.java:166]:
```suggestion
```

File: modules/redpanda/src/test/java/org/testcontainers/redpanda/RedpandaContainerTest.java [hunk 4]
```diff
@@ -138,6 +139,90 @@ public void testUsageWithListener() throws Exception {
...
+            Map<String, String> params = new HashMap<>();
+            params.put("username", username);
+            params.put("password", password);
+            params.put("algorithm", algorithm);
+
+            RestAssured.given().contentType("application/json").body(params).post(adminUrl).then().statusCode(200);
+
+            kcat.start();
+
+            // produceConsumeMessage {
```
Comment (eddumelendez) [inline modules/redpanda/src/test/java/org/testcontainers/redpanda/RedpandaContainerTest.java:180]:
```suggestion
```

File: modules/redpanda/src/test/java/org/testcontainers/redpanda/RedpandaContainerTest.java [hunk 5]
```diff
@@ -138,6 +139,90 @@ public void testUsageWithListener() throws Exception {
...
+                    "-X",
+                    "sasl.password=" + password,
+                    "-C",
+                    "-t",
+                    "msgs",
+                    "-c",
+                    "1"
+                )
+                .getStdout();
+            // }
```
Comment (eddumelendez) [inline modules/redpanda/src/test/java/org/testcontainers/redpanda/RedpandaContainerTest.java:220]:
```suggestion
```

**Review by eddumelendez — COMMENTED** [review]
Thanks for the contribution, @lburgazzoli! Those block are needed for our docs but using existing one will break the documentation.

File: modules/redpanda/src/main/java/org/testcontainers/redpanda/RedpandaContainer.java [hunk 6]
```diff
@@ -232,6 +232,7 @@ private Transferable getRedpandaFile(Configuration cfg) {
                     Map<String, Object> listenerMap = new HashMap<>();
                     listenerMap.put("address", listener.getAddress());
                     listenerMap.put("port", listener.getPort());
+                    listenerMap.put("authentication_method", authenticationMethod);
```
Comment (eddumelendez) [inline modules/redpanda/src/main/java/org/testcontainers/redpanda/RedpandaContainer.java:235]:
```suggestion
                    listenerMap.put("authentication_method", this.authenticationMethod);
```

**Review by eddumelendez — APPROVED** [review]

**Comment by eddumelendez** [comment]
Thanks again, @lburgazzoli! This is now merged in `main` branch and will be part of the next release.

**Comment by lburgazzoli** [comment]
@eddumelendez thx for fixing up my mistakes

**Pull request merged**